### PR TITLE
Add interactor view and pass CoreData to physics models

### PR DESCRIPTION
**Pull request by sethrj**
Another step in the direction of refactoring as Actions and modifying core data:

- Remove `ModelData` and `ModelRef` in favor of `CoreRef` and pass these to kernels
- Add `ids` structs to all the interactors
- Add `InteractionLauncher` to wrap and simplify interactor kernels

### CODE REVIEW

File: src/sim/CoreTrackView.hh [hunk 1]
```diff
@@ -213,5 +224,17 @@ CELER_FUNCTION auto CoreTrackView::make_secondary_allocator() const
     return SecondaryAllocator{states_.secondaries};
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Return a secondary stack allocator view.
+ */
+CELER_FUNCTION auto CoreTrackView::make_relaxation_helper(ElementId el_id) const
```
Comment (amandalund) [inline src/sim/CoreTrackView.hh:231]: Update description.

File: test/physics/em/UrbanMsc.test.cc [hunk 2]
```diff
@@ -304,5 +304,5 @@ TEST_F(UrbanMscTest, msc_scattering)
                                      -0.98020130119347};
 
     EXPECT_VEC_SOFT_EQ(expected_fstep, fstep);
-    EXPECT_VEC_SOFT_EQ(expected_angle, angle);
+    EXPECT_VEC_SOFT_NEAR(expected_angle, angle, 1e-10);
```
Comment (amandalund) [inline test/physics/em/UrbanMsc.test.cc:307]:
```suggestion
    EXPECT_VEC_NEAR(expected_angle, angle, 1e-10);
```

**Review by amandalund — APPROVED** [review]
Awesome @sethrj! Nice code reduction, the launchers look so much cleaner!

File: test/physics/em/UrbanMsc.test.cc (hunk 2 above)
Comment (sethrj) [reply test/physics/em/UrbanMsc.test.cc]: Yep... noticed that it wasn't building when vecgeom was disabled. I'm going to open another small PR to make this change, and to build (but don't run) when vecgeom is disabled, and to do a few more MSC tweaks.

**Pull request merged**